### PR TITLE
[feat] added video actions (like, watch later and history)

### PR DESCRIPTION
**Pull request by omjadhav85**
Features added:
1. User can now like/dislike videos.
2. User can now mark/unmark video as watch later.
3. Track user's watch history. User can delete indivudual items or clear entire history at once
4. This data will retain for each seperate user even after user logs out and logs in.
5. Horizontal Video card

### CODE REVIEW

**Review by hinanshisuthar — COMMENTED** [review]
Good code written. Functionalities work as intended. Nice logic implementation throughout. Good to merge!

File: src/components/VideoCard/VideoCard.js [hunk 1]
```diff
@@ -24,7 +37,20 @@ export const VideoCard = ({ video }) => {
...
+              onClick={() =>
+                authState.token
+                  ? toggleWatchLater(
+                      video,
+                      dispatchData,
+                      authState.token,
+                      isWatchLater
+                    )
+                  : navigate("/login")
+              }
```
Comment (adith-a-danthi) [inline src/components/VideoCard/VideoCard.js]: Click Handler can be moved into a separate function

**Review by adith-a-danthi — COMMENTED** [review]
Clean well structured code. Works as indicated.
One suggestion is to move some of the large blocks of code in `onClick` into separate functions.

File: src/components/VideoCard/VideoCard.js (hunk 1 above)
Comment (omjadhav85) [reply src/components/VideoCard/VideoCard.js]: Made the change....thanks!!

**Comment by omjadhav85** [comment]
Thanks Hinanshi and Adith! Merging this PR after two reviews.

**Pull request merged**